### ISSUE TITLE
Support local_batch and local_subset in tornado netapi

### ISSUE THREAD

**Issue by mchugh19** [saltstack/salt#40476]
### Description of Issue/Question
Currently sending `client: local_batch` or `client: local_subet` to salt-api running tornado returns 500, but works with cherrypy

```
# salt --versions
Salt Version:
           Salt: 2016.11.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.5.0
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-70-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by Ch3LL** [comment]
@mchugh19 can you share a sanitized version of your configs to help with test case? Also looks like someone ran into this here https://github.com/saltstack/salt/issues/32574 and fixed it because of a bad config setup. Can you verify indents and configuration is correct. Thanks

**Comment by mchugh19** [comment]
Sure. 

Here's the response from tornado:
```
 curl -i localhost:8001
HTTP/1.1 200 OK
Date: Mon, 03 Apr 2017 13:43:08 GMT
Content-Length: 99
Etag: "aeff1152c7d469b8706b951c530e056620ee4624"
Content-Type: application/json
Server: TornadoServer/4.2.1

{"clients": ["runner", "runner_async", "local_async", "local"], "return": "Welcome"}
```

Which does not mention local_batch or local_subet.

Here's the response from cherrypy:
```
# curl -i localhost:8000
HTTP/1.1 200 OK
Content-Length: 146
Access-Control-Expose-Headers: GET, POST
Vary: Accept-Encoding
Server: CherryPy/3.5.0
Allow: GET, HEAD, POST
Access-Control-Allow-Credentials: true
Date: Mon, 03 Apr 2017 13:44:16 GMT
Access-Control-Allow-Origin: *
Content-Type: application/json
Set-Cookie: session_id=916950b3fca120dc6ccc2c2991c938dec1b80d53; expires=Mon, 03 Apr 2017 23:44:16 GMT; Path=/

{"clients": ["local", "local_async", "local_subset", "runner", "runner_async", "ssh", "wheel", "wheel_async"], "return": "Welcome"}
```
(which also does not mention local_batch as this was removed in [2016.11.2](https://docs.saltstack.com/en/latest/topics/releases/2016.11.2.html)


When attempting to use the unsupported methods, tornado salt-api returns an error:
```
# cat batch.sh
#!/bin/sh

port=8001

JSON='{"client":"local_batch","batch":"100%","tgt":"*","expr_form":"glob","fun":"cmd.run","arg":["ls -la"],"kwarg":{}}'

RESPONSE=`curl -ik http://localhost:${port}/login -H 'Content-Type: application/json' -d '{"eauth":"pam","password":"jenkins","username":"jenkins"}' 2>/dev/null`
TOKEN=$(echo $RESPONSE |
    sed -e 's/[{}]/''/g' |
    sed -e 's/[{}]/''/g' | sed -e 's/"return"://g' |
    awk -v k="text" '{n=split($0,a,","); for (i=1; i<=n; i++) print a[i]}' |
    grep token | cut -d: -f 2 | cut -d'"' -f 2 )

RESPONSE=`curl -k http://localhost:${port}/ -H 'Accept: application/json' -H "X-Auth-Token: $TOKEN" -H 'Content-Type: application/json' -d "$JSON" 2>/dev/null`
echo $RESPONSE
```
```
#tornado
# bash batch.sh
{"return": ["Unexpected exception while handling request: global name 'get_batch_size' is not defined"]}
```

```
#Using cherrypy with https://github.com/saltstack/salt/commit/3d8f3d18f6afa760c70db87cbbaaa71d877ca4d3 reverted
# bash batch.sh
{"return": [{"minion1": "total 20\ndrwx------ 3 root root 4096 Mar 25 07:54 .\ndrwxr-xr-x 24 root root 4096 Mar 30 06:39 ..\n-rw-r--r-- 1 root root 3106 Oct 22 2015 .bashrc\n-rw-r--r-- 1 root root 148 Aug 17 2015 .profile\ndrwx------ 2 root root 4096 Mar 25 07:54 .ssh"}, {"master": "total 20\ndrwx------ 3 root root 4096 Mar 25 07:51 .\ndrwxr-xr-x 25 root root 4096 Mar 30 06:40 ..\n-rw-r--r-- 1 root root 3106 Oct 22 2015 .bashrc\n-rw-r--r-- 1 root root 148 Aug 17 2015 .profile\ndrwx------ 2 root root 4096 Mar 25 07:51 .ssh"}]}
```


Basically, tornado is missing support for local_batch and local_subset. Cherrypy used to support local_batch, but due to a security need, took it out until it could be corrected (#40477). So this is simply a request to bring tornado up to the same support level as cherrypy.

**Comment by whiteinge** [comment]
You're right that Saltnado [doesn't yet have support for subset](https://github.com/saltstack/salt/blob/595888a439767ebd2cd8c7737e2036cdd6a98572/salt/netapi/rest_tornado/saltnado.py#L252-L256).

It's missing a few other features as well that we're tracking over in #26505. I'm going to close this issue in deference to that one. Thanks for filing this though -- I added a list of missing client interfaces to that other issue just now so that it will show up in search results 😛 . And we can track the reintroduction of batch support in your other issue, #40477.
